composables/api/useLoginUser.ts

```diff
@@ -0,0 +1,57 @@
+import { ref } from 'vue'
+import { useRuntimeConfig } from '#app'
+import { useCurrentUser, useSessionToken } from '@/composables'
+import { LoginRequest, ProblemDetails } from '@/lib/api/data-contracts'
+import { Users } from '@/lib/api/Users'
+
+interface LoginUserResponse {
+  loading: boolean
+  error: ProblemDetails | null
+}
+
+export const useLoginUser = (requestData: LoginRequest): LoginUserResponse => {
+  const responseLoading = ref(true)
+  const responseError = ref<ProblemDetails | null>(null)
+  const authToken = useSessionToken()
+  const currentUser = useCurrentUser()
+
+  const userClient = new Users({
+    baseUrl: useRuntimeConfig().public.BACKEND_BASE_URL,
+  })
+  userClient
+    .usersLoginCreate(requestData)
+    .then((response) => {
+      if (response.ok) {
+        authToken.value = response.data.token
+      } else {
+        throw response.error
+      }
+    })
+    .then(() => {
+      userClient
+        .usersMeList({
+          headers: { Authorization: `Bearer ${authToken.value}` },
+        })
+        .then((response) => {
+          if (response.ok) {
+            currentUser.value = response.data
+          } else {
+            throw response.error
+          }
+        })
+
+      responseLoading.value = false
+    })
+    .catch((error: ProblemDetails) => {
+      console.error(error) // eslint-disable-line no-console
+      responseLoading.value = false
+      responseError.value = error
+    })
+
+  return {
+    error: responseError.value,
+    loading: responseLoading.value,
+  }
+}
+
+export default useLoginUser
```

Comment:
From what I've noticed it seems like most projects are trying to move away from using default exports, Is it reasonable for us to move this to be a named export in this case?

Reply:
Why the move away from default exports if I could ask?

Reply:
I'm also curious

Reply:
It seems like all Libraries and Repos are moving to that to prevent unecessary imports mostly from what it seems is to prevent muddiness as changes come and go knowing exactly what you are importing. The only default export that seems to stay is for things like components so like importing a vue component to be used in a template.

I'm not saying we have to, just definitely a thing i've seen libaries and frameworks do over the past few years.

Reply:
I see. In that case I'd say we can keep this default export as we're only exporting this function. If we change what gets exported in the future, then we'll update accordingly.